services/controllerhost/queueDepth.go

```diff
@@ -325,66 +340,53 @@ func (qdc *queueDepthCalculator) calculateAndReportMetrics(dstDesc *shared.Desti
 			extrapolatedTime,
 			maxExtrapolation)
 
-		bu := common.ExtrapolateDifference(
-			common.SequenceNumber(rs.GetLastSequence()),
-			common.SequenceNumber(rs.GetAvailableSequence()),
-			rs.GetLastSequenceRate(),
-			rs.GetAvailableSequenceRate(),
-			common.UnixNanoTime(rs.GetWriteTime()),
-			common.UnixNanoTime(rs.GetWriteTime()),
-			extrapolatedTime,
-			maxExtrapolation)
+		bu := int64(0) // Unavailable backlog is only applicable to timer destinations, which are deprecated
 
 		if isTabulationRequested {
 			qdc.ll.WithFields(bark.Fields{
-				`time`:                extrapolatedTime,
-				common.TagDst:         common.FmtDst(dstDesc.GetDestinationUUID()),
-				common.TagDstPth:      dstDesc.GetPath(),
-				common.TagCnsm:        cgDesc.GetConsumerGroupUUID(),
-				common.TagCnsPth:      cgDesc.GetConsumerGroupName(),
-				common.TagExt:         string(extent),
-				common.TagStor:        string(store),
-				`rsStore`:             rs.GetStoreUUID(),
-				`rsAvailSeq`:          rs.GetAvailableSequence(),
-				`rsLastSeq`:           rs.GetLastSequence(),
-				`rsLastSeqRate`:       rs.GetLastSequenceRate(),
-				`cgeAckLvlSeq`:        consumerGroupExtent.GetAckLevelSeqNo(),
-				`rsAvailSeqRate`:      rs.GetAvailableSequenceRate(),
-				`cgeAckLvlSeqRate`:    consumerGroupExtent.GetAckLevelSeqNoRate(),
-				`rsWriteTime`:         (extrapolatedTime - common.UnixNanoTime(rs.GetWriteTime())).ToSecondsFmt(),
-				`cgeWriteTime`:        (extrapolatedTime - common.UnixNanoTime(consumerGroupExtent.GetWriteTime())).ToSecondsFmt(),
-				`subTotalAvail`:       ba,
-				`runningTotalAvail`:   ba + BacklogAvailable,
-				`subTotalUnavail`:     bu,
-				`runningTotalUnavail`: bu + BacklogUnavailable,
+				`time`:              extrapolatedTime,
+				common.TagDst:       common.FmtDst(dstDesc.GetDestinationUUID()),
+				common.TagDstPth:    dstDesc.GetPath(),
+				common.TagCnsm:      cgDesc.GetConsumerGroupUUID(),
+				common.TagCnsPth:    cgDesc.GetConsumerGroupName(),
+				common.TagExt:       string(extent),
+				common.TagStor:      string(store),
+				`rsAvailSeq`:        storeExtent.seqAvailable,
+				`cgeAckLvlSeq`:      consumerGroupExtent.GetAckLevelSeqNo(),
+				`rsAvailSeqRate`:    storeExtent.seqAvailableRate,
+				`cgeAckLvlSeqRate`:  consumerGroupExtent.GetAckLevelSeqNoRate(),
+				`rsWriteTime`:       (extrapolatedTime - common.UnixNanoTime(storeExtent.writeTime)).ToSecondsFmt(),
+				`rsCacheTime`:       (extrapolatedTime - common.UnixNanoTime(storeExtent.cacheTime.UnixNano())).ToSecondsFmt(),
+				`cgeWriteTime`:      (extrapolatedTime - common.UnixNanoTime(consumerGroupExtent.GetWriteTime())).ToSecondsFmt(),
+				`subTotalAvail`:     ba,
+				`runningTotalAvail`: ba + BacklogAvailable,
 			}).Info(`Queue Depth Tabulation`)
 		}
 
 		BacklogAvailable += ba
 		BacklogUnavailable += bu
 	}
 
-	coverMap = qdc.destinationExtentsCache.getExtentCoverMap()
+	coverMap = qdc.getCoverMap(false)
```

Comment:
The existing code deletes entries from the cover map. An explicit copy must be made so that multiple consumer groups under a destination can have a mutable copy of the extent map.